precompiles/vesting/integration_test.go

```diff
@@ -647,157 +767,181 @@ var _ = Describe("Interacting with the vesting extension", func() {
+					differentAddr.Bytes(),
+					math.NewInt(1e18),
+				)
 				Expect(err).ToNot(HaveOccurred(), "error while funding the account: %v", err)
-
-				updateFunderArgs := s.BuildCallArgs(callType, contractAddr).
-					WithMethodName(vesting.UpdateVestingFunderMethod).
-					WithPrivKey(differentPriv).
-					WithArgs(
-						s.address,
-						differentAddr,
-						toAddr,
-					)
+				Expect(s.network.NextBlock()).To(BeNil())
+
+				callArgs, txArgs := s.BuildCallArgs(callType, contractAddr)
+				callArgs.MethodName = vesting.UpdateVestingFunderMethod
+				callArgs.Args = []interface{}{
+					s.keyring.GetAddr(0),
+					differentAddr,
+					s.keyring.GetAddr(vestingAccIdx),
+				}
 
 				updateFunderCheck := execRevertedCheck
 				if callType.directCall {
 					updateFunderCheck = failCheck.
 						WithErrContains(fmt.Sprintf(
 							"tx origin address %s does not match the funder address %s",
-							differentAddr.String(), s.address.String(),
+							differentAddr.String(), s.keyring.GetAddr(0).String(),
 						))
 				}
 
-				_, _, err = contracts.CallContractAndCheckLogs(s.ctx, s.app, updateFunderArgs, updateFunderCheck)
-				Expect(err).To(HaveOccurred(), "error while calling the contract: %v", err)
-				if callType.directCall {
-					Expect(err.Error()).To(ContainSubstring("does not match the funder address"))
-				}
+				_, _, err = s.factory.CallContractAndCheckLogs(differentPriv, txArgs, callArgs, updateFunderCheck)
+				Expect(err).NotTo(HaveOccurred(), "error while calling the contract: %v", err)
+				Expect(s.network.NextBlock()).To(BeNil())
+
 				// Check that the vesting account still has the same funder
-				s.ExpectVestingFunder(toAddr, s.address)
+				s.ExpectVestingFunder(s.keyring.GetAddr(vestingAccIdx), s.keyring.GetAddr(0))
 			})
 
 			It(fmt.Sprintf("should return an error when the account does not exist (%s)", callType.name), func() {
 				// Check that there's no account
 				nonExistentAddr := testutiltx.GenerateAddress()
-				acc := s.app.AccountKeeper.GetAccount(s.ctx, nonExistentAddr.Bytes())
+				acc := s.network.App.AccountKeeper.GetAccount(s.network.GetContext(), nonExistentAddr.Bytes())
 				Expect(acc).To(BeNil(), "expected no account to be found")
 
-				updateFunderArgs := s.BuildCallArgs(callType, contractAddr).
-					WithMethodName(vesting.UpdateVestingFunderMethod).
-					WithArgs(
-						s.address,
-						differentAddr,
-						nonExistentAddr, // the address of the vesting account
-					)
+				callArgs, txArgs := s.BuildCallArgs(callType, contractAddr)
+				callArgs.MethodName = vesting.UpdateVestingFunderMethod
+				callArgs.Args = []interface{}{
+					s.keyring.GetAddr(0),
+					differentAddr,
+					nonExistentAddr, // the address of the vesting account
+				}
 
 				updateFunderCheck := execRevertedCheck
 				if callType.directCall {
 					updateFunderCheck = failCheck.
 						WithErrContains(fmt.Sprintf(
-							"account %s does not exist",
+							"account at address '%s' does not exist",
```

Comment:
nit: is the same as `%q`

```suggestion
							"account at address %q does not exist",
```

Reply:
this breaks the test because is using double quotes instead of single quotes